mqtt/handler_test.go

```diff
@@ -0,0 +1,373 @@
+package mqtt
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/mqtt/mocks"
+	mqttredis "github.com/mainflux/mainflux/mqtt/redis"
+	"github.com/mainflux/mainflux/pkg/errors"
+	"github.com/mainflux/mainflux/pkg/messaging"
+	"github.com/mainflux/mainflux/pkg/uuid"
+	"github.com/mainflux/mproxy/pkg/session"
+	"github.com/ory/dockertest/v3"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	thingID        = "thingID"
+	clientID       = "clientID"
+	invalidThingID = "invalidThingID"
+	password       = "password"
+)
+
+var buf bytes.Buffer
+var sessionClient = session.Client{
+	ID:       clientID,
+	Username: thingID,
+	Password: []byte(password),
+}
+
+func TestAuthConnect(t *testing.T) {
+	handler := newHandler()
+
+	cases := []struct {
+		desc    string
+		err     error
+		session *session.Client
+	}{
+		{
+			desc:    "connect without active session",
+			err:     errMissingClient,
+			session: nil,
+		},
+		{
+			desc: "connect with valid id",
```

Comment:
`connect with invalid password"` 

Maybe you can add another test with empty client ID?